tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -158,16 +269,56 @@ static void rx_xmit_adv(void)
 	};
 	int err;
 
-	k_sem_init(&observer_sem, 0, 1);
-	err = bt_le_scan_start(&scan_param, xmit_scan_cb);
+	err = bt_le_scan_start(&scan_param, receive_order_scan_cb);
 	ASSERT_FALSE(err && err != -EALREADY, "starting scan failed (err %d)", err);
 
-	err = k_sem_take(&observer_sem,
-		K_MSEC(((xmit_param->retr + 1) * xmit_param->interval) + 10));
-	ASSERT_OK(err, "Didn't receive adv in time");
+	previous_checker = 0xff;
+	for (int i = 0; i < expect_adv; i++) {
+		err = k_sem_take(&observer_sem, K_SECONDS(10));
+		ASSERT_OK(err, "Didn't receive adv in time");
+	}
 
 	err = bt_le_scan_stop();
 	ASSERT_FALSE(err && err != -EALREADY, "stopping scan failed (err %d)", err);
+
+	PASS();
+}
+
+static void send_adv_buf(struct net_buf *buf, uint8_t curr, uint8_t prev)
+{
+	send_cb.start = send_order_start_cb;
+	send_cb.end = send_order_end_cb;
+
+	(void)net_buf_add_u8(buf, curr);
+	(void)net_buf_add_u8(buf, prev);
+
+	bt_mesh_adv_send(buf, &send_cb, buf);
+	net_buf_unref(buf);
+}
+
+static void send_all_adv_buf(struct net_buf **buf, int num_buf, bool reverse)
```

Comment:
Why int and not size_t for num_buf? ARRAY_SIZE() evaluates to size_t. Unsigned type would mean you can drop the explicit typecasts too, I think.